### DISCUSSION TITLE
How to use ERL_MAX_PORTS and verify the setting in rabbitmq

### DISCUSSION THREAD

**Discussion by sakshiag** [rabbitmq/rabbitmq-server#9311] [Questions] [answered]
We are moving to UBI9 for rabbitmq version 3.12. We read in the rabbitmq docs that UBI9 changed the default open file handlers to a huge number. That creates problems within Erlang. As a result the amount of file handlers should be configured lower. By adjusting ERL_MAX_PORTS to a certain number we can handle this.

Erlang Verison : 26.2
Rabbitmq Version : 3.12
Image :UBI9

We have two questions related to this :
1.  How can we set  ERL_MAX_PORTS as an environment variable. Is there any example to do this ?
2. How can we verify if it is using the set value ?  

Can you please help us with the above queries?

**Comment by michaelklishin** [comment, marked as answer]
1. The same way you would set any other environment variable for any other process, including [using systemd unit files](https://www.flatcar.org/docs/latest/setup/systemd/environment-variables/) if RabbitMQ is installed via an RPM. There is nothing RabbitMQ-specific
2. `rabbitmq-diagnostics status` and/or `rabbitmqctl eval 'erlang:system_info(port_limit).'`. Note that the limit will be set to the next power of 2, not to the exact `ERL_MAX_PORTS` value, so if set to 8191 it will actually be set to 8192, if set to 8193 it will actually be set to 16384, and so on

**Comment by michaelklishin** [comment]
If this is in response for [this blog post](https://blog.rabbitmq.com/posts/2022/08/high-initial-memory-consumption-of-rabbitmq-nodes-on-centos-stream-9/), note that you have [other options of controlling the open file handle limit](https://rabbitmq.com/networking.html#open-file-handle-limit) on Linux, and on modern [systemd-based distributions](https://rabbitmq.com/install-rpm.html#kernel-resource-limits) specifically.
